src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -824,8 +831,12 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
 
             List<List<NodeItem>> nodeStacks = NodeAutoCompleteUtilities.ComputeNodePlacementHeuristics(clusterConnections, clusterNodes);
 
+            //node to connect to from query node
+            var entryNodeId = ClusterResultItem.Topology.Nodes.ToList()[ClusterResultItem.EntryNodeIndex].Id;
```

Comment:
maybe an extra check in case of inconsistent results from the service ? e.g the list might be empty